Paper.tex

```diff
@@ -1596,6 +1596,13 @@ \subsection{zkSNARK Related Precompiled Contracts}
 \end{eqnarray}
 We define $G_2$ to be the subgroup of $(C_2,+)$ generated by $P_2$. $G_2$ is known to be the only cyclic group of order $q$ on $C_2$. For a point $P$ in $G_2$, we define $\log_{P_2}(P)$ be the smallest natural number $n$ satisfying $n\cdot P_2=P$. With this definition, $\log_{P_2}(P)$ is at most $q-1$.
 
+Let $G_T$ be the multiplicative abelian group underlying $F_{p^{12}}$. It is known that a non-degenerate bilinear map $e : G_1\times G_2 \to G_T$ exists.
+This bilinear map is a type three pairing. The ZK-Snark verification contract takes a set of points $a_i$ that are in $G_1$, and a set of points $b_i$ that are in $G_2$. It checks that
+\begin{equation}
+\prod_{i=0}^{n}e\left(a_i, b_i\right) = 1
+\end{equation}
+If this is true, the contract returns $1$, otherwise it returns $0$, unless any of the points $a_i$ or $b_i$ are not on their respective curves, in which case the transaction is reverted.
+
```

Comment:
Please merge this description and the existing description around line 1644.

Reply:
Does the second commit I made address this satisfactorily?